src/hotspot/share/compiler/compilerDirectives.hpp

```diff
@@ -34,11 +34,11 @@
 
   //      Directives flag name,    type, default value, compile command name
   #define compilerdirectives_common_flags(cflags) \
-    cflags(Enable,                  bool, false, X) \
-    cflags(Exclude,                 bool, false, X) \
-    cflags(BreakAtExecute,          bool, false, X) \
-    cflags(BreakAtCompile,          bool, false, X) \
-    cflags(Log,                     bool, LogCompilation, X) \
+    cflags(Enable,                  bool, false, Enable) \
+    cflags(Exclude,                 bool, false, Exclude) \
```

Comment:
CompileCommand uses a different semantic than CompilerDirectives to exclude methods from compilation. Are you sure you will be preserving backwards compatibility?

See compilerDirectives.cpp lines 333-360 to see how the backwards compatibility is preserved.

Reply:
@neliasso, I don't realize that user can get same effect by setting -XX:CompileCommand=exclude/log/compileonly etc util today.  now I get it why you place X here. `DirectiveSet::compilecommand_compatibility_init` ignore those cflags whose cc_flags are X. 

I withdraw Enable/Exclude/Log but leave BreakAtCompile and BreakAtExecution. 
because it's pointless to support -XX:CompileCommand=option,xxx,log. I prefer to have one single way to do it.